pkg/ipcache/ipcache.go

```diff
@@ -653,14 +643,10 @@ func (ipc *IPCache) deleteLocked(ip string, source source.Source) (namedPortsCha
 		namedPortsChanged = namedPortsChanged && ipc.needNamedPorts.Load()
 	}
 
-	if newHostIP != nil {
-		nodeID = ipc.AllocateNodeID(newHostIP)
```

Comment:
by the way I suspect this allocateNodeID from the delete path is the bug most often hit. Shouldn't have this been a deleteNodeID?

Reply:
I pushed your commit (with a new commit description) to fix this.